packages/preset-mini/src/utils/handlers/handlers.ts

```diff
@@ -97,7 +97,7 @@ function bracketWithType(str: string, type?: string) {
   if (str && str.startsWith('[') && str.endsWith(']')) {
     let base: string | undefined
 
-    const match = str.match(/^\[(color|length):/i)
+    const match = str.match(new RegExp(`^\\[\(${type}\):`, 'i'))
```

Comment:
This will create a new regex in every match which will hurt the performance, I think it's better to leave it as literal for now (probably also better to hoist it)